src/app/user-controls/form-error/form-error.directive.ts

```diff
@@ -69,14 +69,15 @@ export class FormErrorDirective implements OnDestroy, OnChanges {
 
   getStandardErrorMessage(error: ValidationError): string {
     const label = this.fieldLabel || 'Input'
+    const errorDetails = this.fieldControl?.getError(error) // {requiredLength: number, minLength: number}
 
     switch (error) {
       case 'required':
         return `${label} is required`
       case 'minlength':
-        return `${label} must be at least 2 characters`
+        return `${label} must be at least ${errorDetails?.requiredLength ?? 2} characters`
       case 'maxlength':
-        return `${label} can\'t exceed 50 characters`
+        return `${label} can\'t exceed ${errorDetails?.requiredLength ?? 50} characters`
```

Comment:
`??` is the cool new TypeScript nullish operator.  

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing

Reply:
@brendoncaulkins yup, updated Ch7 with that info